vscode/oso/syntaxes/polar.tmLanguage.json

```diff
@@ -218,7 +218,7 @@
       }
     },
     "rule-functor": {
-      "begin": "([a-zA-Z][a-zA-Z0-9-_]*)\\(",
+      "begin": "([a-zA-Z_][a-zA-Z0-9_]*(?:::[a-zA-Z0-9_]+)*)\\s*\\(",
```

Comment:
TODO (not in this PR): figure out a way to split this grammar up so we can maintain a single instance of this "identifier" regex as something like `#identifier` and then reference it in rule (name), variable, specializer, and resource block type.

Reply:
The changes in this PR allow this to be highlighted correctly:

```
    _::_  (
  _::_      :    _::_
        {
    _
     :

    _

     }


     )
       ;
```

Reply:
<img width="269" alt="image" src="https://user-images.githubusercontent.com/17556281/147703531-de1e54a4-7e11-48f4-bcad-a1f4a756fed6.png">

Reply:
Lol Gabe what is this madness

Reply:
(I get it, just find it funny you went for something that is completely unintelligible as opposed to doing something like `my::rule(with::var: and::specializer)`)

Reply:
it's space invaders